packages/network-controller/src/NetworkController.ts

```diff
@@ -173,7 +173,7 @@ export type NetworkControllerMessenger = RestrictedControllerMessenger<
 export type NetworkControllerOptions = {
   messenger: NetworkControllerMessenger;
   trackMetaMetricsEvent: () => void;
-  infuraProjectId?: string;
+  infuraProjectId: string;
```

Comment:
Sorry I should elaborate; the path to removing these would be to enhance the capabilities of our custom RPC configuration, such that it can be used to support our Infura networks as well.

We've already started down this path; the localhost e2e test network used to be baked-in, but now we use it as a custom network instead. And when we recently added Linea as a default network, it was added as a custom network.

That is, I'd expect us to delete the built-in providers completely rather than make them optional. Making them optional isn't a step along the path to removing them.

It's long past time to track this work in an issue; I'll set a reminder to create one tomorrow.

Reply:
Issue created: https://github.com/MetaMask/core/issues/1279